test/extended/compatibility.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+#
+# Runs extended compatibility tests with a previous version
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+
+# For now, compatibility tests will not require a node
+# so tests can execute quicker
+SKIP_NODE=1
+source "${OS_ROOT}/test/extended/setup_compat.sh"
+
+os::test::extended::setup_compatibility
+os::test::extended::focus "$@"
+
+function join { local IFS="$1"; shift; echo "$*"; }
+
+focus="\[Compatibility\]"
+
+exitstatus=0
+
+echo "[INFO] Running compatibility tests"
+TEST_REPORT_FILE_NAME=compatibility ${GINKGO} -v "-focus=${focus}" ${EXTENDEDTEST} -- -ginkgo.v -test.timeout 2h || exitstatus=$?
```

Comment:
I'm not a huge fan of the "pattern" where one function sets up a bunch of env vars and others can consume them since it's hard to trace where things go and how they're used -- this seems like a small API that could be exposed as a function in `test/extended/setup_compat.sh`?